{"name": "test_links_of_main_directions", "status": "failed", "statusDetails": {"message": "selene.core.exceptions.TimeoutException: Message: \n\nTimed out after 4s, while waiting for:\nbrowser.element(('css selector', '[data-test=\"menu_section_my_ivi\"]')).click\n\nReason: ElementClickInterceptedException: element click intercepted: Element <a data-test=\"menu_section_my_ivi\" href=\"https://www.ivi.tv/\" title=\"Мой Иви\" class=\"nbl-button nbl-button_style_sairo nbl-button_size_stryker nbl-button_hasBadge_0\">...</a> is not clickable at point (470, 36). Other element would receive the click: <div class=\"segmentedLanding__headerContent\">...</div>\n  (Session info: chrome=128.0.6613.137)\nScreenshot: file:///Users/maxnovo/.selene/screenshots/1743413805870/1743413805870.png\nPageSource: file:///Users/maxnovo/.selene/screenshots/1743413805870/1743413805870.html", "trace": "fn = <selene.core.wait.Command object at 0x104d7da90>\n\n    def logic(fn: Callable[[E], R]) -> R:\n        finish_time = time.time() + self._timeout\n    \n        while True:\n            try:\n>               return fn(self.entity)\n\n.venv/lib/python3.12/site-packages/selene/core/wait.py:119: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n.venv/lib/python3.12/site-packages/selene/core/wait.py:61: in __call__\n    return self._fn(entity)\n.venv/lib/python3.12/site-packages/selene/core/entity.py:525: in raw_click\n    element.locate().click()\n.venv/lib/python3.12/site-packages/selenium/webdriver/remote/webelement.py:119: in click\n    self._execute(Command.CLICK_ELEMENT)\n.venv/lib/python3.12/site-packages/selenium/webdriver/remote/webelement.py:572: in _execute\n    return self._parent.execute(command, params)\n.venv/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py:429: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x103b33c50>\nresponse = {'status': 400, 'value': '{\"value\":{\"error\":\"element click intercepted\",\"message\":\"element click intercepted: Element ...\\\\n#20 0x558845aa5013 \\\\u003Cunknown>\\\\n#21 0x558845ab4ebc \\\\u003Cunknown>\\\\n#22 0x7f6b74a63ac3 \\\\u003Cunknown>\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.ElementClickInterceptedException: Message: element click intercepted: Element <a data-test=\"menu_section_my_ivi\" href=\"https://www.ivi.tv/\" title=\"Мой Иви\" class=\"nbl-button nbl-button_style_sairo nbl-button_size_stryker nbl-button_hasBadge_0\">...</a> is not clickable at point (470, 36). Other element would receive the click: <div class=\"segmentedLanding__headerContent\">...</div>\nE         (Session info: chrome=128.0.6613.137)\nE       Stacktrace:\nE       #0 0x558845ab60ca <unknown>\nE       #1 0x5588457822b0 <unknown>\nE       #2 0x5588457d87e6 <unknown>\nE       #3 0x5588457d66ed <unknown>\nE       #4 0x5588457d4187 <unknown>\nE       #5 0x5588457d302c <unknown>\nE       #6 0x5588457c6946 <unknown>\nE       #7 0x5588457f6902 <unknown>\nE       #8 0x5588457c62f9 <unknown>\nE       #9 0x5588457f6ace <unknown>\nE       #10 0x558845814e26 <unknown>\nE       #11 0x5588457f66a3 <unknown>\nE       #12 0x5588457c44f3 <unknown>\nE       #13 0x5588457c54fe <unknown>\nE       #14 0x558845a7cd7b <unknown>\nE       #15 0x558845a80d31 <unknown>\nE       #16 0x558845a68c85 <unknown>\nE       #17 0x558845a818a2 <unknown>\nE       #18 0x558845a4de1f <unknown>\nE       #19 0x558845aa4e48 <unknown>\nE       #20 0x558845aa5013 <unknown>\nE       #21 0x558845ab4ebc <unknown>\nE       #22 0x7f6b74a63ac3 <unknown>\n\n.venv/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py:232: ElementClickInterceptedException\n\nDuring handling of the above exception, another exception occurred:\n\n    @allure.tag(\"web\")\n    @allure.label(\"owner\", \"Max Razzhivin\")\n    @allure.feature(\"Здесь проверяем основные разделы главной страницы - что по ним можно перейти\")\n    @allure.story(\"При входе на главную страницу мы можем перейти на другие разделы сайта \"\n                  \"- 'Мой иви', 'Фильмы', 'Сериалы', 'Мультфильмы'\")\n    @allure.link(\"https://www.ivi.ru/\", \"Ivi.ru main page\")\n    @allure.severity(Severity.CRITICAL)\n    def test_links_of_main_directions():\n        no_login=NotAuthorised()\n        with allure.step(\"Открываем главную страницу\"):\n            open_page()\n        with allure.step(\"Проверяем переходы по разделам - Мой иви, фильмы, сериалы и мультфильмы\"):\n>           no_login.links_to_main_directions()\n\ntests/test_selene_ivi.py:42: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\npages/main_page.py:103: in links_to_main_directions\n    browser.element('[data-test=\"menu_section_my_ivi\"]').click()\n.venv/lib/python3.12/site-packages/selene/core/entity.py:540: in click\n    self.wait.for_(\n.venv/lib/python3.12/site-packages/selene/core/wait.py:150: in for_\n    return self._decorator(self)(logic)(fn)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nfn = <selene.core.wait.Command object at 0x104d7da90>\n\n    def logic(fn: Callable[[E], R]) -> R:\n        finish_time = time.time() + self._timeout\n    \n        while True:\n            try:\n                return fn(self.entity)\n            except Exception as reason:\n                if time.time() > finish_time:\n                    reason_string = '{name}: {message}'.format(\n                        name=reason.__class__.__name__,\n                        message=getattr(reason, \"msg\", str(reason)),\n                    )\n                    # TODO: think on how can we improve logging failures in selene, e.g. reverse msg and stacktrace\n                    # stacktrace = getattr(reason, 'stacktrace', None)\n                    # TODO: should we have an option to turn on stacktrace logging?\n                    timeout = self._timeout\n                    entity = self.entity\n    \n                    # if it's a normal function, it should have __qualname__,\n                    # then use it, otherwise use str(fn)\n                    fn_name = getattr(\n                        fn,\n                        '__qualname__',\n                        getattr(fn, '__str__', lambda: str(fn))(),\n                    )\n    \n                    failure = TimeoutException(\n                        f'\\n'\n                        f'\\nTimed out after {timeout}s, while waiting for:'\n                        f'\\n{entity}.{fn_name}'\n                        f'\\n'\n                        f'\\nReason: {reason_string}'\n                    )\n    \n>                   raise self._hook_failure(failure)\nE                   selene.core.exceptions.TimeoutException: Message: \nE                   \nE                   Timed out after 4s, while waiting for:\nE                   browser.element(('css selector', '[data-test=\"menu_section_my_ivi\"]')).click\nE                   \nE                   Reason: ElementClickInterceptedException: element click intercepted: Element <a data-test=\"menu_section_my_ivi\" href=\"https://www.ivi.tv/\" title=\"Мой Иви\" class=\"nbl-button nbl-button_style_sairo nbl-button_size_stryker nbl-button_hasBadge_0\">...</a> is not clickable at point (470, 36). Other element would receive the click: <div class=\"segmentedLanding__headerContent\">...</div>\nE                     (Session info: chrome=128.0.6613.137)\nE                   Screenshot: file:///Users/maxnovo/.selene/screenshots/1743413805870/1743413805870.png\nE                   PageSource: file:///Users/maxnovo/.selene/screenshots/1743413805870/1743413805870.html\n\n.venv/lib/python3.12/site-packages/selene/core/wait.py:148: TimeoutException"}, "steps": [{"name": "Открываем главную страницу", "status": "passed", "start": 1743413843637, "stop": 1743413850280}, {"name": "Проверяем переходы по разделам - Мой иви, фильмы, сериалы и мультфильмы", "status": "failed", "statusDetails": {"message": "selene.core.exceptions.TimeoutException: Message: \n\nTimed out after 4s, while waiting for:\nbrowser.element(('css selector', '[data-test=\"menu_section_my_ivi\"]')).click\n\nReason: ElementClickInterceptedException: element click intercepted: Element <a data-test=\"menu_section_my_ivi\" href=\"https://www.ivi.tv/\" title=\"Мой Иви\" class=\"nbl-button nbl-button_style_sairo nbl-button_size_stryker nbl-button_hasBadge_0\">...</a> is not clickable at point (470, 36). Other element would receive the click: <div class=\"segmentedLanding__headerContent\">...</div>\n  (Session info: chrome=128.0.6613.137)\nScreenshot: file:///Users/maxnovo/.selene/screenshots/1743413805870/1743413805870.png\nPageSource: file:///Users/maxnovo/.selene/screenshots/1743413805870/1743413805870.html\n\n", "trace": "  File \"/Users/maxnovo/Desktop/qa.guru/Test_Project_Ivi/tests/test_selene_ivi.py\", line 42, in test_links_of_main_directions\n    no_login.links_to_main_directions()\n  File \"/Users/maxnovo/Desktop/qa.guru/Test_Project_Ivi/pages/main_page.py\", line 103, in links_to_main_directions\n    browser.element('[data-test=\"menu_section_my_ivi\"]').click()\n  File \"/Users/maxnovo/Desktop/qa.guru/Test_Project_Ivi/.venv/lib/python3.12/site-packages/selene/core/entity.py\", line 540, in click\n    self.wait.for_(\n  File \"/Users/maxnovo/Desktop/qa.guru/Test_Project_Ivi/.venv/lib/python3.12/site-packages/selene/core/wait.py\", line 150, in for_\n    return self._decorator(self)(logic)(fn)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/maxnovo/Desktop/qa.guru/Test_Project_Ivi/.venv/lib/python3.12/site-packages/selene/core/wait.py\", line 148, in logic\n    raise self._hook_failure(failure)\n"}, "start": 1743413850281, "stop": 1743413867167}], "start": 1743413843636, "stop": 1743413867176, "uuid": "49c57a28-c61f-4411-97b4-9274eca1c391", "historyId": "3acd9b5ab64c2f1beea965bcdc002e1d", "testCaseId": "3acd9b5ab64c2f1beea965bcdc002e1d", "fullName": "tests.test_selene_ivi#test_links_of_main_directions", "labels": [{"name": "feature", "value": "Здесь проверяем основные разделы главной страницы - что по ним можно перейти"}, {"name": "story", "value": "При входе на главную страницу мы можем перейти на другие разделы сайта - 'Мой иви', 'Фильмы', 'Сериалы', 'Мультфильмы'"}, {"name": "tag", "value": "web"}, {"name": "severity", "value": "critical"}, {"name": "owner", "value": "Max Razzhivin"}, {"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_selene_ivi"}, {"name": "host", "value": "MacBook-Air-MaxNovo.local"}, {"name": "thread", "value": "5705-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_selene_ivi"}], "links": [{"type": "Ivi.ru main page", "url": "https://www.ivi.ru/", "name": "https://www.ivi.ru/"}]}